bin_heap.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+
+from __future__ import unicode_literals
+
+
+class BinaryHeap(object):
+    def __init__(self, itr=None):
+        self.heap = []
+        self.size = 0
+        if itr is not None:
+            for val in itr:
+                self.push(val)
+
+    def push(self, val):
+        """adds to the end of the list and bubbles up to the right
+        position.
+        """
+        self.heap.append(val)
+        self.size += 1
+        self.heapify_up(self.size-1)
```

Comment:
pep8: e226 missing whitespace around arithmetic operator. Also to be consistent with style used on other lines such as line 24.